src/umpire/resource/FileMemoryResource.cpp

```diff
@@ -52,23 +50,22 @@ void* FileMemoryResource::allocate(std::size_t bytes)
   if (fd == -1) { UMPIRE_ERROR("Opening File Failed: " << strerror(errno)); }
 
   // Setting Size Of Map File
-  std::size_t SIZE = bytes / sysconf(_SC_PAGE_SIZE);
-  if(SIZE == 0)
-    SIZE = sysconf(_SC_PAGE_SIZE);
+  std::size_t num_pages = bytes / sysconf(_SC_PAGE_SIZE);
```

Comment:
Perhaps we could consider replacing this and lines 54-48 with:
```suggestion
  const std::size_t pagesize{ sysconf(_SC_PAGE_SIZE) };
  std::size_t rounded_bytes{ ( ( bytes + (pagesize - 1) ) / pagesize ) * pagesize};
```

Then use `rounded_bytes` for `ftruncate` and `mmap` and std::pair calls.